script.js

```diff
@@ -1,54 +1,72 @@
+/* eslint-disable max-classes-per-file */
+/* eslint-disable class-methods-use-this */
+
 const container = document.querySelector('.container');
 const titleInput = document.querySelector('#title');
 const authorInput = document.querySelector('#author');
 const submit = document.querySelector('#add-button');
 const form = document.querySelector('.form');
-function Books(title, author) {
-  this.title = title;
-  this.author = author;
+
+class Cards {
+  constructor() {
+    this.array = [];
+  }
+
+  AddCard(book) {
+    this.array.push(book);
+  }
 }
-let cards = [];
-function ShowCards() {
-  const removeDivs = document.querySelectorAll('.card');
-  removeDivs.forEach((div) => {
-    div.remove();
-  });
-  cards.forEach((card) => {
-    const div = document.createElement('div');
-    div.classList.add('card');
-    const heading = document.createElement('h1');
-    const paragraph = document.createElement('p');
-    const deleteBook = document.createElement('button');
-    deleteBook.classList.add('delete-book');
-    heading.textContent = `${card.title}`;
-    paragraph.textContent = `${card.author}`;
-    deleteBook.textContent = 'Remove';
-    div.appendChild(heading);
-    div.appendChild(paragraph);
-    div.appendChild(deleteBook);
-    container.appendChild(div);
-
-    deleteBook.addEventListener('click', (e) => {
-      cards.splice(cards.indexOf(card), 1);
-      e.target.parentNode.remove();
-      const dataMarker = JSON.stringify(cards);
-      localStorage.setItem('data', dataMarker);
+
+const collection = new Cards();
+
+class Books {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+
+  ShowCards() {
+    const removeDivs = document.querySelectorAll('.card');
+    removeDivs.forEach((div) => {
+      div.remove();
     });
-  });
+    collection.array.forEach((card) => {
+      const div = document.createElement('div');
+      div.classList.add('card');
+      const paragraph = document.createElement('p');
+      const deleteBook = document.createElement('button');
+      deleteBook.classList.add('delete-book');
+      paragraph.textContent = `"${card.title}" by ${card.author}`;
+      deleteBook.textContent = 'Remove';
+      div.appendChild(paragraph);
+      div.appendChild(deleteBook);
+      container.appendChild(div);
+      deleteBook.addEventListener('click', (e) => {
+        collection.array.splice(collection.array.indexOf(card), 1);
+        e.target.parentNode.remove();
+        const dataMarker = JSON.stringify(collection.array);
+        localStorage.setItem('data', dataMarker);
+      });
+    });
+  }
 }
 
 function addBook(title, author) {
   const book = new Books(title, author);
```

Comment:
- [ ] Please make sure the methods for adding and removing books are class-level and should be inside the class as requested in the project requirements. Currently, your methods are outside of the class.

![image](https://github.com/Zilola-Nazarova/awesome_books/assets/83803414/21f45506-ae5a-49e3-9458-7bd5782a2c8d)

Reply:
Thank you @thecodechaser . We've decided to move the call for the methods and constructors to the eventlistener, because we already have the method "ShowCards" that adds the book to the collection inside of the class.